framework/logresource/resource.go

```diff
@@ -0,0 +1,145 @@
+package logresource
+
+import (
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name                 = "log"
+	PostOperationMessage = "executed resource operation without errors"
+	PreOperationMessage  = "start to execute resource operation"
+)
+
+// Config represents the configuration used to create a new log resource.
+type Config struct {
+	// Dependencies.
+	Logger   micrologger.Logger
+	Resource framework.Resource
+}
+
+// DefaultConfig provides a default configuration to create a new log resource
+// by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		Logger:   nil,
+		Resource: nil,
+	}
+}
+
+type Resource struct {
+	// Dependencies.
+	logger   micrologger.Logger
+	resource framework.Resource
+}
+
+// New creates a new configured log resource.
+func New(config Config) (*Resource, error) {
+	// Dependencies.
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+	if config.Resource == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Resource must not be empty")
+	}
+
+	newResource := &Resource{
+		// Dependencies.
+		logger: config.Logger.With(
+			"underlyingResource", config.Resource.Underlying().Name(),
+		),
+		resource: config.Resource,
+	}
+
+	return newResource, nil
+}
+
+func (r *Resource) GetCurrentState(obj interface{}) (interface{}, error) {
+	r.logger.Log("debug", PreOperationMessage, "operation", "GetCurrentState")
```

Comment:
Using the log resource we can get rid of some log magic we would repeat over and over again. 

> I am thinking to have a log resource which does logging according to information in a context. We could have micrologger.Logger.WithContext(ctx, key, val) and add the key value pairs to the context at the very beginning of the reconciliation loop initialization. The log resource then takes the key-value pairs from the context and adds them to the logging it does. We would then also need to extend the framework interface to take the context.

See also https://github.com/giantswarm/aws-operator/pull/373#discussion_r132698523.

The idea with the context magic is not part of this PR. So lets iterate and see how it goes.